gammapy/background/off_data_background_maker.py

```diff
@@ -0,0 +1,248 @@
+
+    def group_observations(self):
+        """Group the background observation list.
+
+        For now we'll just use the zenith binning from HAP.
+        """
+
+        obs_table = self.define_obs_table()
+
+        # Define observation groups
+        # zenith_bins = np.array([0, 20, 30, 40, 50, 90])
+        zenith_bins = np.array([0, 49, 90])
+        # zenith_bins = np.array([0, 30, 90])  # for testing
+        axes = [ObservationGroupAxis('ZEN_PNT', zenith_bins, fmt='edges')]
+        obs_groups = ObservationGroups(axes)
+        log.info(obs_groups.info)
+
+        # Apply observation grouping
+        obs_table = obs_groups.apply(obs_table)
+
+        # Store the results
+        filename = self.obs_table_grouped_filename
+        log.info('Writing {}'.format(filename))
+        obs_table.write(str(filename), format='ascii.ecsv')
+
+        filename = self.group_table_filename
+        log.info('Writing {}'.format(filename))
+        obs_groups.obs_groups_table.write(str(filename), format='ascii.ecsv')
+        self.ntot_group = obs_groups.n_groups
+
+    def make_model(self, modeltype, obs_table=None, excluded_sources=None):
+        """Make background models.
+
+        Create the list of background model (`~gammapy.background.CubeBackgroundModel` (3D) or
+        `~gammapy.background.EnergyOffsetBackgroundModel` (2D)) for each group in zenithal angle and efficiency
+
+        Parameters
+        ----------
+        modeltype : str
+            type of the background modelisation: 3D or 2D
+        obs_table : `~astropy.table.Table`
+            observation table of the OFF run List used for the background modelling
+        excluded_sources : `~astropy.table.Table`
+            Table of excluded sources.
+            Required columns: RA, DEC, Radius
+        """
+
+        if not obs_table:
+            filename = self.obs_table_grouped_filename
+            log.info('Reading {}'.format(filename))
+            obs_table = ObservationTable.read(str(filename), format='ascii.ecsv')
+
+        groups = sorted(np.unique(obs_table['GROUP_ID']))
+        log.info('Groups: {}'.format(groups))
+        for group in groups:
+            # Get observations in the group
+            idx = np.where(obs_table['GROUP_ID'] == group)[0]
+            obs_table_group = obs_table[idx]
+            log.info('Processing group {} with {} observations'.format(group, len(obs_table_group)))
+
+            # Build the model
+            if modeltype == "3D":
+                model = CubeBackgroundModel.define_cube_binning(obs_table_group, method='default')
+                model.fill_obs(obs_table_group, self.data_store)
+                model.smooth()
+                model.compute_rate()
+                self.models3D.append(model)
+
+            elif modeltype == "2D":
+                ebounds = EnergyBounds.equal_log_spacing(0.1, 100, 100, 'TeV')
+                offset = sqrt_space(start=0, stop=2.5, num=100) * u.deg
```

Comment:
I changed to a sqrt distribution.
Maybe we have to give the energy and offset as parameters no? This is weird to define them here?